### PR TITLE
Performance

### LINKED ISSUE #164: fsr/semester/x and results/semester/x need 10 seconds to load
fsr/semester/x and results/semester/x need 10 seconds to load.

### PR DESCRIPTION
**Pull request by karyon**
this PR improves performance on the user and semester index pages significantly and a little bit on all other pages as well. fixes #164. it also does minor cleanups and moves stuff around in the settings.

### CODE REVIEW

File: evap/staff/fixtures/minimal_test_data.json [hunk 1]
```diff
@@ -1181,6 +1181,58 @@
   "pk": 36,
   "model": "evaluation.contribution",
   "fields": {
+    "contributor": 2,
```
Comment (janno42) [inline evap/staff/fixtures/minimal_test_data.json]: why do we need all the extra test data?

**Comment by janno42** [comment]
please fix the tests

File: evap/staff/fixtures/minimal_test_data.json (hunk 1 above)
Comment (karyon) [inline evap/staff/fixtures/minimal_test_data.json]: they were inconsistent before. multiple courses had no responsible contribution and/or general contribution.

**Comment by karyon** [comment]
test is fixed.

**Comment by janno42** [comment]
:+1:

**Pull request merged**